lua/telescope/pickers.lua

```diff
@@ -716,8 +716,14 @@ function Picker:delete_selection(delete_cb)
     return x > y
   end)
   for _, index in ipairs(selection_index) do
-    local selection = table.remove(self.finder.results, index)
-    delete_cb(selection)
+    -- WIP: Trying to implement logic which will not remove the item from selection if buffer deletion failed
+    --
+    local selection = self.finder.results[index]
+    local no_error, function_return = pcall(delete_cb, selection)
```

Comment:
hmm we can do this but we should not do a interface around pcall.

We can say that `delete_cb` returns either `nil` or a `boolean`. If its `true` or `nil` (so we dont break backwards compat) we remove it otherwise we dont

Reply:
Will gladly implement it this way, but would like to understand the logic behind the backwards compatibility argument.
I didn't find that the `delete_cb` is run within `pcall` somewhere else.

What would break if the exception gets cough at that point?
Would it not propagate correctly to the upper layers and that's the compatibility your're worried about?

(full disclosure, I come mostly from Python in terms for how I think exceptions work, from what I read, lua is similar in the sense that exceptions propagate up the stack if uncouth. If something I say doesn't make sense, it's likely coming from that incorrect mental model)

Reply:
lua works the same, there is just no reason here to build an interface around passing exceptions up the stack. You already caught the exception in the callback.

`delete_selection` can be used outside of this project as well for other pickers. We dont have to keep backwards compatibility and have the callback be `function(selection) -> boolean` only, its still better than only building this interface on exceptions

Reply:
Thanks

> We dont have to keep backwards compatibility and have the callback be function(selection) -> boolean only, its still better than only building this interface on exceptions

I guess the docs also need a change (defining the return values [here](https://github.com/nvim-telescope/telescope.nvim/blob/ee6111952a8c76b65e59b26a7efe62efb1998590/lua/telescope/pickers.lua#L693)) as that's the way to communicate the interface of the callback function?